### PR TITLE
Formating (indentation) backend/dwarfdbginfo.d

### PR DESCRIPTION
**Pull request by Luhrel**

### CODE REVIEW

File: src/dmd/backend/dwarfdbginf.d [hunk 1]
```diff
@@ -51,1532 +53,1514 @@ else
...
+            uint total_length;
+            ushort version_;
+            uint abbrev_offset;
+            ubyte address_size;
+        }
 
-AApair *type_table;
-AApair *functype_table;  // not sure why this cannot be combined with type_table
-Outbuffer *functypebuf;
+        static assert(DebugInfoHeader.alignof == 1 && DebugInfoHeader.sizeof == 11);
```
Comment (ibuclaw) [inline src/dmd/backend/dwarfdbginf.d:566]: Would be good to have this in a separate commit saying that baseline compiler is now 2.076.

File: src/dmd/backend/dwarfdbginf.d [hunk 2]
```diff
@@ -51,1532 +53,1514 @@ else
...
-        buf.writeByten(8);
+        // Length of CIE, not including padding
+        const uint cielen = 4 + 4 + 1 +
+            (ehunwind ? 5 : 3) +
+            1 + 1 + 1 +
+            (ehunwind ? 8 : 2) +
+            5;
 
-        buf.writeByten(DW_CFA_offset + 16);    // OFFSET r16,1   RIP is at -8*1[RSP]
-        buf.writeByten(1);
```
Comment (ibuclaw) [inline src/dmd/backend/dwarfdbginf.d:773]: Removing this line is not doing anyone any favours.

File: src/dmd/backend/dwarfdbginf.d [hunk 3]
```diff
@@ -51,1532 +53,1514 @@ else
...
-    }
-    else
-    {
-        buf.writeByten(DW_CFA_def_cfa);        // DEF_CFA ESP,4
-        buf.writeByten(cast(ubyte)dwarf_regno(SP));
-        buf.writeByten(4);
-
-        buf.writeByten(DW_CFA_offset + 8);     // OFFSET r8,1
-        buf.writeByten(1);
```
Comment (ibuclaw) [inline src/dmd/backend/dwarfdbginf.d:782]: Nor this line.

File: src/dmd/backend/dwarfdbginf.d [hunk 4]
```diff
@@ -51,1532 +53,1514 @@ else
...
 
-        __gshared DebugFrameFDE64 debugFrameFDE64 =
-        {   20,             // length
-            0,              // CIE_pointer
-            0,              // initial_location
-            0,              // address_range
-        };
+            buf.writeByten(DW_CFA_offset + 8);     // OFFSET r8,1
+        }
+        buf.writeByten(1);
```
Comment (ibuclaw) [inline src/dmd/backend/dwarfdbginf.d:776]: Because now it might not be clear what this is for.

File: src/dmd/backend/dwarfdbginf.d [hunk 5]
```diff
@@ -51,1532 +53,1514 @@ else
...
+                debug_line.buf.writeByte(0);
+            }
 
-    if (infoFileName_table)
-    {
-        AAchars.destroy(infoFileName_table);
-        infoFileName_table = null;
-    }
+        version (MARS)
+        version (none)
```
Comment (ibuclaw) [inline src/dmd/backend/dwarfdbginf.d:1068]: I think the one liner is a more readable, but that's just stylistic opinion.

File: src/dmd/backend/dwarfdbginf.d [hunk 6]
```diff
@@ -51,1532 +53,1514 @@ else
...
-    debug_info.initialize();
-
-    debuginfo = debuginfo_init;
-    if (I64)
-        debuginfo.address_size = 8;
-
-    // Workaround https://issues.dlang.org/show_bug.cgi?id=16563
-    // Struct alignment is ignored due to 2.072 regression.
-    static if (debuginfo.alignof == 1)
+        assert(debuginfo.alignof == 1);
```
Comment (ibuclaw) [inline src/dmd/backend/dwarfdbginf.d:1115]: Same here regarding separate commit because of adjusted baseline compiler comment.

File: src/dmd/backend/dwarfdbginf.d [hunk 7]
```diff
@@ -51,1532 +53,1514 @@ else
...
+        debug_info.buf.writeByte(0);      // ending abbreviation code
 
-    debug_info.buf.writeByte(0);      // ending abbreviation code
+        debuginfo.total_length = cast(uint)debug_info.buf.length() - 4;
 
-    debuginfo.total_length = cast(uint)debug_info.buf.length() - 4;
-    // Workaround https://issues.dlang.org/show_bug.cgi?id=16563
-    // Struct alignment is ignored due to 2.072 regression.
-    static if (debuginfo.alignof == 1)
+        assert(debuginfo.alignof == 1);
```
Comment (ibuclaw) [inline src/dmd/backend/dwarfdbginf.d:1449]: And here too (baseline compiler updated)

**Review by ibuclaw — CHANGES_REQUESTED** [review]
As a rule of thumb, refactorings like this shouldn't be adding or removing any lines.  Just making what is available more uniformly consistent in style.

File: src/dmd/backend/dwarfdbginf.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/backend/dwarfdbginf.d:566]: Actually, a separate PR would be even better.

Comment (Luhrel) [reply src/dmd/backend/dwarfdbginf.d:566]: I don't get it.
What does the future person who reads this line benefits with a comment saying "Baseline compiler is now 2.076" ?
I'll add the bug link, I think it's more relevant.

Doing a separate PR.

Comment (ibuclaw) [reply src/dmd/backend/dwarfdbginf.d]: Commit message, not comment.

**Review by ibuclaw — APPROVED** [review]
